pandas/core/indexes/base.py

```diff
@@ -871,7 +871,7 @@ def _format_data(self, name=None) -> str_t:
         if self.inferred_type == "string":
             is_justify = False
         elif self.inferred_type == "categorical":
-            if is_object_dtype(self.categories):  # type: ignore
```

Comment:
FYI, I had to remove this due to mypy complaints.

Reply:
Tests failed due to this line. However, `mypy` run as part of git-commit complains here saying "unused 'type: ignore' comment".